### PR TITLE
Another attempt to refactor aggregate report query model.

### PR DESCRIPTION
**Pull request by agorina**
Here is another try to make it a little cleaner. Most of the changes are due to renaming. I will point out real changes.

Note: that this broke the communication with UI. Will need @esotrope to help me with getting it working.

### CODE REVIEW

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/model/CustomAggregateOrganizationQuery.java [hunk 1]
```diff
@@ -10,7 +10,7 @@
  * Represents query parameters for the custom aggregate repository.
  */
 @SuppressWarnings("unchecked")
-public class AggregateReportQuery extends OrganizationTypeQuery {
+public class CustomAggregateOrganizationQuery extends OrganizationTypeQuery {
```
Comment (agorina) [inline aggregate-service/src/main/java/org/opentestsystem/rdw/olap/model/CustomAggregateOrganizationQuery.java]: Renamed it due to the name conflicts.

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/model/LongitudinalOrganizationQuery.java [hunk 2]
```diff
@@ -10,7 +10,7 @@
  * Represents query parameters for the Longitudinal repository.
  */
 @SuppressWarnings("unchecked")
-public class LongitudinalReportQuery extends OrganizationTypeQuery {
+public class LongitudinalOrganizationQuery extends OrganizationTypeQuery {
```
Comment (agorina) [inline aggregate-service/src/main/java/org/opentestsystem/rdw/olap/model/LongitudinalOrganizationQuery.java]: Ditto.

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/service/impl/DefaultBasicCustomAggregateService.java [hunk 3]
```diff
@@ -39,7 +39,7 @@
 
     @Override
     public boolean handles(final AggregateQuery query) {
-        return Basic.equals(query.getQueryType());
+        return CustomAggregate.equals(query.getReportType()) && !query.isFilteredSubgroupQuery();
```
Comment (agorina) [inline aggregate-service/src/main/java/org/opentestsystem/rdw/olap/service/impl/DefaultBasicCustomAggregateService.java]: This is a real change: now we have a report type and a flag for filtered subgroups.

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/service/impl/DefaultBasicLongitudinalService.java [hunk 4]
```diff
@@ -39,7 +39,7 @@
 
     @Override
     public boolean handles(final AggregateQuery query) {
-        return BasicLongitudinal.equals(query.getQueryType());
+        return Longitudinal.equals(query.getReportType()) && !query.isFilteredSubgroupQuery();
```
Comment (agorina) [inline aggregate-service/src/main/java/org/opentestsystem/rdw/olap/service/impl/DefaultBasicLongitudinalService.java]: The change again.

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/service/impl/DefaultFilteredSubgroupAggregateService.java [hunk 5]
```diff
@@ -39,7 +40,7 @@
 
     @Override
     public boolean handles(final AggregateQuery query) {
-        return FilteredSubgroup.equals(query.getQueryType());
+        return CustomAggregate.equals(query.getReportType()) && query.isFilteredSubgroupQuery();
```
Comment (agorina) [inline aggregate-service/src/main/java/org/opentestsystem/rdw/olap/service/impl/DefaultFilteredSubgroupAggregateService.java]: And one more change.

File: common/src/main/java/org/opentestsystem/rdw/reporting/common/model/AbstractAggregateReportQuery.java [hunk 6]
```diff
@@ -27,6 +32,30 @@
     private String achievementLevelDisplayType;
     private ImmutableList<String> columnOrder;
 
+    //defines custom subgroups
```
Comment (agorina) [inline common/src/main/java/org/opentestsystem/rdw/reporting/common/model/AbstractAggregateReportQuery.java]: I combined FilteredSubgroups and Basic queries into one and added a method to return if it is a filtered subgroup.

File: common/src/main/java/org/opentestsystem/rdw/reporting/common/model/AggregateQuery.java [hunk 7]
```diff
@@ -10,7 +10,14 @@
...
      */
-    AggregateQueryType getQueryType();
+    AggregateReportType getReportType();
+
+
+    /**
+     *
+     * @return true if the query includes filtered subgroups
+     */
+    boolean isFilteredSubgroupQuery();
```
Comment (agorina) [inline common/src/main/java/org/opentestsystem/rdw/reporting/common/model/AggregateQuery.java]: This is the main change. Now the backed model is aligned with the front end in a sense that there are two categories: a report type and a flag indicating if it is a basic query or filtered subgroups.

File: aggregate-service/src/test/java/org/opentestsystem/rdw/olap/repository/impl/JdbcCustomAggregateReportRepositoryRST.java [hunk 8]
```diff
@@ -911,7 +911,7 @@ public void itShouldRequireDimension() {
 
     @Test(expected = IllegalArgumentException.class)
     public void itShouldRequireOrganization() {
-        repository.findByOrganizationType(user, AggregateReportQuery
+        repository.findByOrganizationType(user, CustomAggregateOrganizationQuery
```
Comment (malaffoon) [inline aggregate-service/src/test/java/org/opentestsystem/rdw/olap/repository/impl/JdbcCustomAggregateReportRepositoryRST.java]: As i read through i admit to liking the old name better. I haven't seen the reason for the rename but i'm sure i'll run into it.

File: common/src/main/java/org/opentestsystem/rdw/reporting/common/model/AggregateReportType.java [hunk 9]
```diff
@@ -3,15 +3,14 @@
 /**
  * This enum contains the available types of aggregate queries.
  */
-public enum AggregateQueryType {
-    Basic(BasicCustomAggregateReportQuery.class),
-    FilteredSubgroup(FilteredSubgroupCustomAggregateReportQuery.class),
-    BasicLongitudinal(BasicLongitudinalReportQuery.class),
-    FilteredSubgroupLongitudinal(FilteredSubgroupLongitudinalReportQuery.class);
+public enum AggregateReportType {
```
[the remark this answers is not in the document]
Comment (malaffoon) [reply common/src/main/java/org/opentestsystem/rdw/reporting/common/model/AggregateReportType.java]: https://youtu.be/xOrgLj9lOwk?t=1m28s

**Comment by agorina** [comment]
I am going to merge (since @esotrope verbally approved it and helped me with the front end). Will address comments if needed.

File: common/src/main/java/org/opentestsystem/rdw/reporting/common/model/AggregateReportType.java (hunk 9 above)
[the remark this answers is not in the document]
Comment (agorina) [reply common/src/main/java/org/opentestsystem/rdw/reporting/common/model/AggregateReportType.java]: You know I am not the right person to post a comment like this :)
(I just do not have the appreciation I am sure it deservers).

**Pull request merged**